docs/src/main/asciidoc/application-configuration-guide.adoc

```diff
@@ -185,6 +185,41 @@ You can override these runtime properties with the following mechanisms (in decr
 
 NOTE: Environment variables names are following the conversion rules of link:https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/configsources.asciidoc#default-configsources[Eclipse MicroProfile]
 
+=== Configuration Profiles
+
+Quarkus supports the notion of configuration profiles. These allow you to have multiple configuration in the same file and
+select between then via a profile name.
+
+The syntax for this is `%{profile}.config.key=value`. For example if I have the following:
+
+----
+quarkus.http.port=9090
+%dev.quarkus.http.port=8181
+----
+
+The Quarkus HTTP port will be 9090, unless the 'dev' profile is active, in which case it will be 8181.
+
+By default Quarkus has three profiles, although it is possible to use as many as you like. The default profiles are:
+
+* *dev* - Activated when in development mode (i.e. `quarkus:dev`)
+* *test* - Activated when running tests
+* *prod* - The default profile when not running in development or test mode
+
+There are two ways to set a custom profile, either via the `quarkus-profile` system property or the `QUARKUS_PROFILE`
```

Comment:
Wouldn't we want this to be `quarkus.profile` to keep it within our namespace?  When looking at solutions to problems around configuration values which are only given as environment variables, it occurred to me that we'll want to be able to manage the `QUARKUS_` env namespace as well as the `quarkus.` config namespace; introducing ambiguity here might be a problem later, or at least confusing.

Reply:
It won't complain about an unconsumed config property if we use `quarkus.`? It bugged me too but then I remembered we had some issues with it earlier.

Reply:
As long as that property is bound to a field, it'll work.  We could also easily have a mechanism for cases where we don't want a field for whatever reason.

Reply:
This is consumed before any of the normal config stuff. This means that none of the standard things you would expect from config will work such as expansion, or reading from multiple sources.  In addition if we bound this to a field it would require special case treatment in any sort of config file or automatic documentation that we generated, as it is not a real config property in the same way that others are, it is just a system property that has meaning to Quarkus.

There are pros and cons to either one, but I thought that this was least likely to introduce confusion and would not require special case handing in anything that is consuming the config metadata.